extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -401,9 +404,21 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       return;
     }
     this.win['dbclk_a4a_viz_change'] = true;
-    addExperimentIdToElement(
-        isExperimentOn(this.win, 'a4a-safeframe-preloading-off') ?
-        '21061136' : '21061135', this.element);
+
+    const sfPreloadExpName = 'a4a-safeframe-preloading-off';
+    const experimentInfoMap =
+        /** @type {!Object<string, !ExperimentInfo>} */ ({});
+    experimentInfoMap[sfPreloadExpName] = {
+      isTrafficEligible: () => true,
+      branches: ['21061135', '21061136'],
+    };
+    randomlySelectUnsetExperiments(this.win, experimentInfoMap);
```

Comment:
Would be nice if randomlySelectUnsetExperiments returned the selected experiment id (or null) and if addExperimentIdToElement allowed ?string for experiment id (where if null does nothing) so this could be collapsed to:

const sfPreloadExpId = randomlySelectUnsetExperiments(this.win, experimentInfoMap);
addExperimentIdToElement(sfPreloadExpId, this.element);
this.preloadSafeframe_ = sfPreloadExpId == '#';

Perhaps in a later PR?

Reply:
Unfortunately `randomlySelectUnsetExperiments` can set multiple experiments, so it would have to return a list. Still might be worthwhile, though.